app/src/main/java/com/github/freeman/bootcamp/games/guessit/lobbies/LobbyListActivity.kt

```diff
@@ -59,14 +62,33 @@ class LobbyListActivity: ComponentActivity() {
 
         setContent {
             BootcampComposeTheme {
+                val password = remember { mutableStateOf("") }
+                val enterPassword = remember { mutableStateOf(false) }
+                val lobby = remember { mutableStateOf(Lobby("", "", 0, 0, "", "")) }
+
                 Column {
                     TopAppbarLobbies()
 
                     Surface(
                         modifier = Modifier.fillMaxSize(),
                         color = Color(0xFFF1F1F1)
                     ) {
-                        LobbyList(dbRef)
+                        if (!enterPassword.value) {
+                            LobbyList(dbRef, enterPassword, lobby)
+                        }
+                    }
+
+                    // Shows the dialog when a private lobby is clicked
+                    if (enterPassword.value) {
+                        val context = LocalContext.current
+                        val userId = Firebase.auth.uid
+                        EditDialog(password, ENTER_PASSWORD_TEXT, PASSWORD_TEXT, enterPassword) {
```

Comment:
To be coherent with my other comment about the keyboard type, I suggest to also change it here, when the user has to enter a password.